### ISSUE TITLE
Halyard uses endpoint removed in Kubernetes 1.10

### ISSUE THREAD

**Issue by will-gtv** [spinnaker/spinnaker#2728]
### Issue Summary:
`hal deploy apply` uses the old style proxy path for kubernetes which was removed in v1.10

### Cloud Provider(s):
- Kubernetes

### Environment:
Spinnaker running on and deploying to a cluster upgraded to 1.10 in AWS.

### Feature Area: 
Halyard: `hal deploy apply`

### Description:
When deploying services to kubernetes halyard calls `http://localhost:41573/api/v1/proxy/namespaces/spinnaker/services/spin-orca-bootstrap:8083/orchestrate` which gives a 404 in kubernetes 1.10 as the base `/api/v1/proxy` path has been removed in favour of `/api/v1/namespaces/{namespace}/{type}/{object}/proxy`. This shows up as `ERROR Unexpected exception: retrofit.RetrofitError: 404 Not
  Found` and all services failing to deploy.

### Steps to Reproduce:
- Deploy or upgrade to kubernetes 1.10
- Configure halyard to deploy to kubernetes
- Run `hal deploy apply`

### Additional Details:
Kubernetes PR removing the path used: https://github.com/kubernetes/kubernetes/pull/59884

`/var/log/spinnaker/halyard/halyard.log`:
```
---> HTTP POST http://localhost:41573/api/v1/proxy/namespaces/spinnaker/services/spin-orca-bootstrap:8083/orchestrate
<--- HTTP 404 http://localhost:49117/api/v1/proxy/namespaces/spinnaker/services/spin-orca-bootstrap:8083/orchestrate (299ms)
2018-05-01 14:57:57.925 ERROR 32208 --- [     Thread-156] c.n.s.halyard.core.DaemonResponse        : Unknown exception encountered: 

retrofit.RetrofitError: 404 Not Found
        at retrofit.RetrofitError.httpError(RetrofitError.java:40) ~[retrofit-1.9.0.jar:na]
        at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:388) ~[retrofit-1.9.0.jar:na]
        at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240) ~[retrofit-1.9.0.jar:na]
        at com.sun.proxy.$Proxy132.orchestrate(Unknown Source) ~[na:na]
        at com.netflix.spinnaker.halyard.deploy.deployment.v1.DistributedDeployer.lambda$deployServiceWithOrca$8(DistributedDeployer.java:281) ~[halyard-deploy-1.1.0-20180425120405.jar:1.1.0-20180425120405]
        at com.netflix.spinnaker.halyard.deploy.deployment.v1.OrcaRunner.getTaskEndpoint(OrcaRunner.java:64) ~[halyard-deploy-1.1.0-20180425120405.jar:1.1.0-20180425120405]
        at com.netflix.spinnaker.halyard.deploy.deployment.v1.OrcaRunner.monitorPipeline(OrcaRunner.java:53) ~[halyard-deploy-1.1.0-20180425120405.jar:1.1.0-20180425120405]
        at com.netflix.spinnaker.halyard.deploy.deployment.v1.DistributedDeployer.deployServiceWithOrca(DistributedDeployer.java:282) ~[halyard-deploy-1.1.0-20180425120405.jar:1.1.0-20180425120405]
        at com.netflix.spinnaker.halyard.deploy.deployment.v1.DistributedDeployer.lambda$deploy$4(DistributedDeployer.java:213) ~[halyard-deploy-1.1.0-20180425120405.jar:1.1.0-20180425120405]
        at com.netflix.spinnaker.halyard.core.DaemonResponse$StaticRequestBuilder.build(DaemonResponse.java:127) ~[halyard-core-1.1.0-20180425120405.jar:1.1.0-20180425120405]
        at com.netflix.spinnaker.halyard.core.tasks.v1.TaskRepository.lambda$submitTask$1(TaskRepository.java:48) ~[halyard-core-1.1.0-20180425120405.jar:1.1.0-20180425120405]
        at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_162]

2018-05-01 14:57:57.925  WARN 32208 --- [     Thread-156] c.n.s.h.core.tasks.v1.TaskRepository     : Task [Deploy spin-deck] (f06a9258-6365-4d85-9d28-134ff09a86d6) - RUNNING failed with unexpected reason: 
```

**Comment by lwander** [comment]
Should hopefully be as easy as updating the client library... https://github.com/fabric8io/kubernetes-client 

will have to see

**Comment by ezimanyi** [comment]
Addressed by PR linked above; fix included in Halyard 1.2.0.
